stdlib/grpc/src/main/java/org/ballerinalang/net/grpc/nativeimpl/serviceendpoint/Init.java

```diff
@@ -87,22 +85,17 @@
     
     @Override
     public void execute(Context context) {
-        try {
-            Struct serviceEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
-            BMap<String, BValue> endpointConfigStruct = (BMap<String, BValue>) context.getRefArgument(1);
-            Struct serviceEndpointConfig = BLangConnectorSPIUtil.toStruct(endpointConfigStruct);
-            ListenerConfiguration configuration = getListenerConfig(serviceEndpointConfig);
-            ServerConnector httpServerConnector =
-                    HttpConnectionManager.getInstance().createHttpServerConnector(configuration);
+        Struct serviceEndpoint = BLangConnectorSPIUtil.getConnectorEndpointStruct(context);
+        BMap<String, BValue> endpointConfigStruct = (BMap<String, BValue>) context.getRefArgument(1);
+        Struct serviceEndpointConfig = BLangConnectorSPIUtil.toStruct(endpointConfigStruct);
+        ListenerConfiguration configuration = getListenerConfig(serviceEndpointConfig);
+        ServerConnector httpServerConnector =
+                HttpConnectionManager.getInstance().createHttpServerConnector(configuration);
 
-            ServicesRegistry.Builder grpcServicesRegistryBuilder = new ServicesRegistry.Builder();
-            serviceEndpoint.addNativeData(SERVER_CONNECTOR, httpServerConnector);
-            serviceEndpoint.addNativeData(SERVICE_REGISTRY_BUILDER, grpcServicesRegistryBuilder);
-            context.setReturnValues();
-        } catch (Exception ex) {
```

Comment:
This will let any runtime exceptions to propagate to VM. Which is not a good thing to do. This is actually a limitation in current VM, where we can't trigger throw behavior with a Ballerina error struct. Can you create an issue for that? 

Once we fixed that Ideally, we should have catch clause here and should throw ballerina error from here.

Reply:
I was under impression that when we return error struct with the error message, it throws the error message. but it is not throwing the correct error message to user. I will create an issue and share the link

Reply:
Created issue #10216